cargo-dist/src/lib.rs

```diff
@@ -395,13 +396,133 @@ fn write_checksum(checksum: &str, src_path: &Utf8Path, dest_path: &Utf8Path) ->
     Ok(())
 }
 
+// Takes a string in KEY=value environment variable format and
+// parses it into a BTreeMap. The string syntax is sh-compatible, and also the
+// format returned by `env`.
+fn parse_env(env_string: &str) -> DistResult<BTreeMap<&str, &str>> {
+    let mut parsed = BTreeMap::new();
+    for line in env_string.trim_end().split('\n') {
+        let key;
+        let value;
+        match line.split_once('=') {
+            Some(parsed) => (key, value) = parsed,
+            None => {
+                return Err(DistError::EnvParseError {
+                    line: line.to_owned(),
+                })
+            }
+        };
+
+        parsed.insert(key, value);
+    }
+
+    Ok(parsed)
+}
+
+// Takes a BTreeMap of key/value environment variables produced by
+// `brew bundle exec` and decides which ones we want to keep for our own builds.
+// Returns a Vec containing (KEY, value) tuples.
+fn select_brew_env(environment: &BTreeMap<&str, &str>) -> Vec<(String, String)> {
+    let mut desired_env = vec![];
+
+    // Several of Homebrew's environment variables are safe for us to use
+    // unconditionally, so pick those in their entirety.
+    if let Some(value) = environment.get("PKG_CONFIG_PATH") {
+        desired_env.push(("PKG_CONFIG_PATH".to_owned(), value.to_string()))
+    }
+    if let Some(value) = environment.get("PKG_CONFIG_LIBDIR") {
+        desired_env.push(("PKG_CONFIG_LIBDIR".to_owned(), value.to_string()))
+    }
+    if let Some(value) = environment.get("CMAKE_INCLUDE_PATH") {
+        desired_env.push(("CMAKE_INCLUDE_PATH".to_owned(), value.to_string()))
+    }
+    if let Some(value) = environment.get("CMAKE_LIBRARY_PATH") {
+        desired_env.push(("CMAKE_LIBRARY_PATH".to_owned(), value.to_string()))
+    }
+    let mut paths = vec![];
+    // For each listed dependency, add it to the PATH
+    if let Some(optprefix) = environment.get("HOMEBREW_OPT") {
+        // Comma-separated list of all dependencies in the recursive tree
+        if let Some(dependencies) = environment.get("HOMEBREW_DEPENDENCIES") {
+            for dep in dependencies.split(',') {
+                let short_name = dep.split('/').next().unwrap();
```

Comment:
Would be nice to not unwrap here, if possible

Reply:
Unless I'm missing something, it's impossible for `split` on a string to not return at last a first element. Is there a more idiomatic way to say "get me the first element with no unwrap"? I feel like we don't need to manage an impossible `None` scenario.

Reply:
great point! maybe just a comment reminding the reader that split is so nicely behaved, then